osia/installer/downloader/install.py

```diff
@@ -150,5 +150,5 @@ def download_installer(installer_version: str,
     if root.exists() and root.joinpath('openshift-install').exists():
         logging.info('Found installer at %s', root.as_posix())
         return root.joinpath('openshift-install').as_posix()
-    root.mkdir(parents=True)
+    root.mkdir(parents=True, exist_ok=True)
```

Comment:
I'm not sure that this is desired behavior. If the directory exists it means, that there are potentially some files left by previous installation - e.g. installation files or so. The right way should be check if directory exists and fail-fast of course.

Reply:
But on the other hand you're right that current implementation, which relies on the behaviour of `mkdir` is not valid way at all.

Reply:
I thought this path was for the `openshift-install` binary not the full installation? I think I had a network interruption and ended up with the directory created but the binary not downloaded. I can revert this as its probably a highly unusual scenario.

Reply:
You're right, need to pay more attention to the code. Then your change was completely right and I was mistaken, I'm really sorry. For some reason I got the idea that this is the part where the name of cluster is resolved - that's place you want to fail for sure - because if the directory with that name exists, then it is probably due to failed installation of removal and someone should pay attention to it.

Reply:
I'll leave it to you if you'd like to reintroduce this back as I can see it beneficial to the code, the rest is great and I'm again sorry for the confusion.